File: Coding-Practice/March_Starters33/distinctdilem.cpp
#include <bits/stdc++.h>
#include <iostream>
using namespace std;

int main(){
    int t;
    cin >> t;
    while(t--){
        int n;
        cin >> n;
        int sum = 0;
        for(int i = 0; i < n; i++){
            int j;
            cin >> j;
            sum += j;
        }
        float x = (-1+sqrt(1+8*sum))/2;
        x = floor(x);
        cout << x << endl;

    }
    return 0;
}

File: Coding-Practice/March_Starters33/distinctdilem.py
from cmath import sqrt
from math import floor


import math
t = int(input())

for _ in range(t):
    n = int(input())
    l = list(map(int, input().split()))
    sum = 0
    for i in range(len(l)):
        sum += l[i]
    
    x = (-1+ math.sqrt(1+8*sum))//2
    x = math.floor(x)
    print(x)